src/ethRegistrar.ts

```diff
@@ -91,46 +93,17 @@ export function handleNameRenewed(event: NameRenewedEvent): void {
 
 export function handleNameTransferred(event: TransferEvent): void {
   let label = uint256ToByteArray(event.params.tokenId)
-  let registration = new Registration(label.toHex())
-  registration.registrant = event.params.to.toHex()
-  registration.save()
-
+  // Do not create new registration
```

Comment:
Using new here does not overwrite existing objects - if it already exists, it's merged with the existing record. Doing it this way avoids an unnecessary read. The old code should work fine here.

Reply:
The use of `Registration.load` over `new Registration` is correct here because it is used to refer to `registration.registrant`. It's not only semantically wrong to use `new` here, If you use `new Registration` it will try to access non existing location and panics.

```
INFO 3 triggers found in this block for this subgraphthread 'mapping-QmSidhqBvH4YTmmVsjubdurFjW5hajykZwTcdu2KCVmroV-02a4fc1a-82ad-4e39-9143-a783dbf22cf2' panicked at 'called `Result::unwrap()` on an `Err` value: Memory("trying to access region [66..6291622] in memory [0..65536]")', src/libcore/result.rs:1165:5
graph-node_1  | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
graph-node_1  | , block_hash: 0x3582ba83020c2f24c23979980f8f0c0c2e27d1279cb2cb685b89fd8da320e232, block_number: 66, subgraph_id: QmSidhqBvH4YTmmVsjubdurFjW5hajykZwTcdu2KCVmroV, component: SubgraphInstanceManager
graph-node_1  | Jan 02 17:19:22.921 ERRO Subgraph instance failed to run: Failed to process trigger in transaction 0x9c01…ef0b: Mapping terminated before finishing to handle Ethereum event: Transfer(indexed address,indexed address,indexed uint256), code: SubgraphSyncingFailure, id: QmSidhqBvH4YTmmVsjubdurFjW5hajykZwTcdu2KCVmroV, subgraph_id: QmSidhqBvH4YTmmVsjubdurFjW5hajykZwTcdu2KCVmroV, component: SubgraphInstanceManager
graph-node_1  | Jan 02 17:22:28.60
```

Reply:
It turned out that our current mainnet indexing was also failing with the same error

<img width="1257" alt="Screenshot 2020-01-03 at 10 17 12" src="https://user-images.githubusercontent.com/2630/71731342-79719f80-2e12-11ea-95dd-51762204ff52.png">


This logically makes sense. If you look into new registration events, `Registrar.Transfer` gets fired before `Registrar.NameRegistered` hence it's failing at `handleNameRegistered` complaining that `registrationDate` (which gets set at later events) do not exist.

<img width="960" alt="Screenshot 2020-01-03 at 10 19 37" src="https://user-images.githubusercontent.com/2630/71731404-a32ac680-2e12-11ea-9019-c92ddf5cbe41.png">

Reply:
And just to repeat my point again `Registrar.Transfer` is evoked by `_mint` function which is part of OZ ERC721 code https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/ERC721.sol#L265